srcCxx/ShapeType_main.cxx

```diff
@@ -18,6 +20,13 @@
 #include "ShapeType_configurator_default.h"
 #endif
 
+#ifndef STRING_IN
```

Comment:
what is the purpose of this macro definitions?

Reply:
In the dds-rtps repository, OpenDDS define STRING_IN and STRING_INOUT to be .in() and .inout() respectively, to access sample string members.  I just copied the mechanism to keep that open as an option.

But, now that I look at the code here, I am not sure if it is necessary...  I think @mitza-oci or colleague will need to say if it is still required.

Reply:
`shape.color`, which is a C++-mapped IDL string, is passed as a C varargs parameter to a printf-like function below.  C++ class objects may not be passed through varargs (at the very least it's non-portable implementation-defined behavior).  GCC produces a warning and probably not the correct output at runtime.

I'm wondering why it doesn't impact other DDS implementations.

Reply:
...attempting to answer my own question... I wonder if other implementations are using the IDL-to-C language mapping (in whole or in part).  I assumed C++ since this is the `srcCxx` directory, but it doesn't really have to be.  As far as I know all OMG IDL-to-language mappings that target C++ (any version) use a class for this and not a pointer.

Reply:
We support a couple of different c++ language mappings.  
In this particular case, we (coredx_dds) map string to char * and string<bound> to char[bound+1(?)].

It sounds like we do have to deal with some variations in the data type, so I think the answer is that the STRING_IN and STRING_INOUT are useful.

Reply:
Yeah, they are very simple preprocessor shims to take care of differences between mappings.

Another way to handle it would be to use `ostream::operator<<` instead of printf.  Then the type would either be naturally supported by the standard library, or the implementation (either in middleware library code, generated code, or the custom "configurator" code) can provide its own overload.